cardano-cli/src/Cardano/CLI/Compatible/Transaction.hs

```diff
@@ -242,72 +228,77 @@ runCompatibleTransactionCmd
     ) = do
-    sks <- firstExceptT CompatibleWitnessError $ mapM (newExceptT . readWitnessSigningData) witnesses
-
-    allOuts <- firstExceptT CompatibleTxCmdError $ mapM (toTxOutInAnyEra sbe) outs
-
-    certFilesAndMaybeScriptWits <-
-      firstExceptT CompatibleScriptWitnessReadError $
-        readCertificateScriptWitnesses sbe certificates
-
-    certsAndMaybeScriptWits :: [(Certificate era, Maybe (ScriptWitness WitCtxStake era))] <-
-      shelleyBasedEraConstraints sbe $
-        sequence
-          [ fmap
-              (,cswScriptWitness <$> mSwit)
-              ( firstExceptT CompatibleFileError . newExceptT $
-                  readFileTextEnvelope AsCertificate (File certFile)
-              )
-          | (CertificateFile certFile, mSwit) <- certFilesAndMaybeScriptWits
-          ]
-
-    (protocolUpdates, votes) :: (AnyProtocolUpdate era, AnyVote era) <-
-      caseShelleyToBabbageOrConwayEraOnwards
-        ( const $ do
-            prop <- maybe (pure $ NoPParamsUpdate sbe) readUpdateProposalFile mUpdateProposal
-            return (prop, NoVotes)
-        )
-        ( \w -> do
-            prop <- maybe (pure $ NoPParamsUpdate sbe) readProposalProcedureFile mProposalProcedure
-            votesAndWits <-
-              firstExceptT CompatibleVoteError . newExceptT $
-                readVotingProceduresFiles w mVotes
-            votingProcedures <-
-              firstExceptT CompatibleVoteMergeError . hoistEither $
-                mkTxVotingProcedures [(v, vswScriptWitness <$> mSwit) | (v, mSwit) <- votesAndWits]
-            return (prop, VotingProcedures w votingProcedures)
-        )
-        sbe
-
-    let certsRefInputs =
-          [ refInput
-          | (_, Just sWit) <- certsAndMaybeScriptWits
-          , refInput <- maybeToList $ getScriptWitnessReferenceInput sWit
-          ]
-
-        votesRefInputs =
-          [ refInput
-          | VotingProcedures _ (TxVotingProcedures _ (BuildTxWith voteMap)) <- [votes]
-          , sWit <- Map.elems voteMap
-          , refInput <- maybeToList $ getScriptWitnessReferenceInput sWit
-          ]
-
-        proposalsRefInputs =
-          [ refInput
-          | ProposalProcedures _ (TxProposalProcedures _ (BuildTxWith proposalMap)) <- [protocolUpdates]
-          , sWit <- Map.elems proposalMap
-          , refInput <- maybeToList $ getScriptWitnessReferenceInput sWit
-          ]
-
-    validatedRefInputs <-
-      liftEither . first CompatibleTxCmdError . validateTxInsReference $
-        certsRefInputs <> votesRefInputs <> proposalsRefInputs
-    let txCerts = mkTxCertificates sbe certsAndMaybeScriptWits
-
-    -- this body is only for witnesses
-    apiTxBody <-
-      firstExceptT CompatibleTxBodyError $
-        hoistEither $
+    void $ fromEitherCli $ Left Dummy
+
+    shelleyBasedEraConstraints sbe $ do
+      sks <- mapM (fromEitherIOCli . readWitnessSigningData) witnesses
+
+      allOuts <- fromEitherIOCli . runExceptT $ mapM (toTxOutInAnyEra sbe) outs
```

Comment:
adding another function
```haskell
fromEitherTCli = fromEitherIOCli . runExceptT
```
could save us some boilerplate

Reply:
I'm hesitant about this because we want to remove `ExceptT` usage.

Reply:
I don't understand what you mean. We want to remove `ExceptT` usage in `cardano-cli`.  I.e not implement functions in `ExceptT`.

Reply:
We have functions returning a type `ExceptT e m a` in cardano-api. We still need to use them in cardano-cli.